ts/Checkbox/index.tsx

```diff
@@ -31,7 +23,7 @@ export class Checkbox extends MaterialComponent<
 
   public componentDidMount() {
     super.componentDidMount();
-    if (this.control) {
+    if (this.control && !MaterialComponent.isPrerendering) {
```

Comment:
cDM should never run in a server side code by preact's design. Why this check is even required?

Reply:
I depends on the technique you use, many frameworks just render the page either in the browser or in jsdom (Which doesn't have Mutation observer for example).

Reply:
I agree that there are specific tools like `preact-cli` which don't do that.

Reply:
See https://github.com/GoogleChromeLabs/prerender-loader#string-prerendering for example

Reply:
> headless outdated browser

`jsdom` is no browser, it runs in nodejs and it's used fairly often, if you don't want a headless browser. Headless browser would just work fine.

> react/preact has a first class server side rendering

Haven't seen server side rendering for preact yet

Reply:
Preact has `render to string` just like react, why would you need anything else?

Reply:
Oh I couldn't find this because it's in an separate library